vaadin-cdi/src/test/java/com/vaadin/cdi/ViewDestroyTest.java

```diff
@@ -0,0 +1,65 @@
+package com.vaadin.cdi;
+
+import com.vaadin.cdi.internal.Conventions;
+import com.vaadin.cdi.uis.DestroyViewUI;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+
+public class ViewDestroyTest extends AbstractManagedCDIIntegrationTest {
+
+    private String uiId;
+
+    @Deployment(testable = false)
+    public static WebArchive deployment() {
+        return ArchiveProvider.createWebArchive("viewDestroy", DestroyViewUI.class);
+    }
+
+    @Before
+    public void setUp() throws Exception {
+        resetCounts();
+    }
+
+    @Test
+    public void testViewChangeDestroysViewScope() throws Exception {
+        loadView(DestroyViewUI.VIEWSCOPED_VIEW);
+        assertViewDestroyCounts(0);
+        clickAndWait(DestroyViewUI.NAVIGATE_OTHER_BTN_ID);
+        assertViewDestroyCounts(1);
+    }
+
+    @Test
+    public void testNaviagteToSameViewDestroysViewScope() throws Exception {
```

Comment:
At the time i was defined this behavior i didn't know the changes in vaadin 7.6.7, and 7.6.8.
see issues
https://github.com/vaadin/framework/issues/4112
https://github.com/vaadin/framework/issues/7737

CDIViewProvider.getView create a new contextual instance in a new context, so SharedUtil.equals(getCurrentView(), viewWithLongestName) in Navigator#navigateTo always false.

So navigating to the same view with same parameters is not a no-op, as intended. - intended? -
Issue fixed with https://github.com/vaadin/framework/issues/4112 might be broken this way. 
Needs further investigation.